### PR TITLE
feat: typescript port

### PR DESCRIPTION
**Pull request by navinkarkera**

### CODE REVIEW

**Comment by navinkarkera** [comment]
Example run of CI is [here](https://github.com/navinkarkera/short_stuff/pull/1/checks)

File: ts_version/src/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  for (let index = 0; index < 100; index++) {
+    const thing = uuidv4();
+    expect(unSlugify(slugify(thing))).toBe(thing);
+  }
+});
+
+test("e2e shorthand", () => {
+  for (let index = 0; index < 100; index++) {
+    const thing = genShortcodeUuid();
+    console.log(`(${thing}, ${slugify(thing)})`);
```
Comment (Kelketek) [inline ts_version/src/index.test.ts]: Leftover log statement

File: ts_version/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import { v4 as uuidv4, parse as uuidParse } from 'uuid';
+
+/*Pads a sequence of raw bytes to make them the required size of a UUID.
```
Comment (Kelketek) [inline ts_version/src/index.ts]: Could you format comments like this, to match JSDoc standards?

```javascript
/**
* Pads a sequence of raw bytes to make them the required size of a UUID.
*/
```

**Review by Kelketek — CHANGES_REQUESTED** [review]
@navinkarkera Just a couple of changes here, quite minor.

**Pull request merged**